src/rnp/fficli.cpp

```diff
@@ -902,24 +889,36 @@ cli_rnp_generate_key(rnp_cfg_t *cfg, cli_rnp_t *rnp, const char *username)
         (void) fprintf(stderr, "Failed to set hash algorithm.\n");
         goto done;
     }
-    if (rnp_op_generate_set_protection_cipher(genkey, rnp_cfg_getstr(cfg, CFG_KG_PROT_ALG))) {
-        (void) fprintf(stderr, "Failed to set protection algorithm.\n");
-        goto done;
-    }
-    if (rnp_op_generate_set_protection_hash(genkey, rnp_cfg_getstr(cfg, CFG_KG_PROT_HASH))) {
-        (void) fprintf(stderr, "Failed to set protection hash algorithm.\n");
-        goto done;
-    }
-    if (rnp_op_generate_set_protection_iterations(
-          genkey, rnp_cfg_getint(cfg, CFG_KG_PROT_ITERATIONS))) {
-        (void) fprintf(stderr, "Failed to set protection iterations.\n");
-        goto done;
-    }
     if (rnp_op_generate_execute(genkey) || rnp_op_generate_get_key(genkey, &subkey)) {
         (void) fprintf(stderr, "Subkey generation failed.\n");
         goto done;
     }
 
+    // protect
+    rnp_password_cb passcb;
+    void *          passctx;
+    char            password[MAX_PASSWORD_LENGTH];
+    if (rnp->passfp) {
+        passcb = &ffi_pass_callback_file;
+        passctx = rnp->passfp;
+    } else {
+        passcb = &ffi_pass_callback_stdin;
+        passctx = NULL;
+    }
```

Comment:
What do you think about changing this to some FFI function call like rnp_request_password(ffi, char **password), which will use FFI's configuration instead of this (imagine, we'll add more ways to request password from CLI)?

Reply:
I thought about that too since the context is already wrapped up in the ffi object. I'd rather file a separate issue for further consideration though like you suggested.

Reply:
@dewyatt Ok, filed #1036